monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, Sequence, Union
+
+import numpy as np
+import torch
+
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.data.meta_obj import get_track_meta
+from monai.networks.layers.simplelayers import GaussianFilter
+from monai.transforms.post.array import RemoveSmallObjects, SobelGradients
+from monai.transforms.transform import Transform
+from monai.utils import TransformBackends, convert_to_numpy, optional_import
+from monai.utils.type_conversion import convert_to_dst_type, convert_to_tensor
+
+label, _ = optional_import("scipy.ndimage.measurements", name="label")
+disk, _ = optional_import("skimage.morphology", name="disk")
+opening, _ = optional_import("skimage.morphology", name="opening")
+watershed, _ = optional_import("skimage.segmentation", name="watershed")
+
+__all__ = ["CalcualteInstanceSegmentationMap"]
+
+
+class CalcualteInstanceSegmentationMap(Transform):
+    """
+    Process Nuclei Prediction with XY Coordinate Map.
+
+    Args:
+        threshold_overall: threshold the float values of overall gradient map to int 0 or 1 with specified theashold.
+            Defaults to 0.4.
+        min_size: objects smaller than this size are removed. Defaults to 10.
+        sigma: std. Used in `GaussianFilter`. Could be a single value, or `spatial_dims` number of values. Defaults to 0.4.
+        kernel_size: the size of the Sobel kernel. Defaults to 21.
+        radius: the radius of the disk-shaped footprint. Defaults to 2.
+        gaussian: whether need to smooth the image to be applied by the watershed segmentation. Defaults to False.
+        remove_small_objects: whether need to remove some objects in segmentation results and marker. Defaults to False.
+        marker_postprocess_fn: execute additional post transformation on marker. Defaults to None.
+
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        threshold_overall: float = 0.4,
+        min_size: int = 10,
+        sigma: Union[Sequence[float], float, Sequence[torch.Tensor], torch.Tensor] = 0.4,
+        kernel_size: int = 21,
+        radius: int = 2,
+        gaussian: bool = False,
```

Comment:
Maybe this can be similar to `marker_postprocess_fn` to accept `Optional[Callable]`, and with more generalized name like `smoothing_fn`.